pkg/client/unversioned/discovery_client.go

```diff
@@ -0,0 +1,175 @@
+/*
+Copyright 2015 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package unversioned
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/url"
+
+	"k8s.io/kubernetes/pkg/api/unversioned"
+)
+
+type DiscoveryInterface interface {
+	ServerGroupsInterface
+	ServerResourcesInterface
+}
+
+// GroupsInterface has methods for obtaining supported groups on the API server
```

Comment:
> Are they too verbose?

Yes. But, that matches with the style of the rest of the package. In my ideal world, these interfaces wouldn't exist. It's a long rant of mine nicely summarized by [this article](https://medium.com/@rakyll/interface-pollution-in-go-7d58bccec275), so I won't repeat it here. If I were to define an interface, I wouldn't put Interface in it's name.

But more importantly this matches the current style of this package, so I probably wouldn't change it too much.